File: HotkeyAutomation/HotkeyProcessing/EffectData.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace HotkeyAutomation.HotkeyProcessing
{
	/// <summary>
	/// Represents effect data.  Instead of using separate classed and/or inheritance, this is all in one class to make serialization and deserialization easier.
	/// </summary>
	public class EffectData
	{
		#region HttpGet
		public string httpget_url;
		#endregion
		#region BroadLink
		public string broadlink_name;
		public string broadlink_commandName;
		public byte broadlink_repeatCount;
		#endregion
		#region iTach
		public string itach_name;
		public string itach_connectorAddress;
		public string itach_commandShortName;
		public byte itach_repeatCount;
		#endregion
		#region Vera
		public string vera_name;
		public int? vera_deviceNum;
		[JsonConverter(typeof(StringEnumConverter))]
		public VeraService? vera_service;
		public string vera_value;
		#endregion
		#region HomeAssistant
		public string hass_servername;
		public string hass_entityid;
		[JsonConverter(typeof(StringEnumConverter))]
		public HomeAssistantMethod? hass_method;
		public string hass_value;
		#endregion
	}
	public enum VeraService
	{
		DimmerValue,
		SwitchSet,
		CurtainStop
	}
	public enum HomeAssistantMethod
	{
		DimmerValue,
		SwitchSet,
		CoverStop,
		CoverSet
	}
}

File: HotkeyAutomation/www/scripts/EffectData.js
export var EffectType = {
	HttpGet: "HttpGet",
	BroadLink: "BroadLink",
	iTach: "iTach",
	Vera: "Vera",
	HomeAssistant: "HomeAssistant"
};
export var VeraService = {
	DimmerValue: "DimmerValue",
	SwitchSet: "SwitchSet",
	CurtainStop: "CurtainStop"
};
export var HomeAssistantMethod = {
	DimmerValue: "DimmerValue",
	SwitchSet: "SwitchSet"
};
export class Effect
{
	constructor()
	{
		this.type = EffectType.HttpGet;
		this.data = new EffectData();
	}
}
export class EffectData
{
	constructor()
	{
		this.httpget_url = null;
		this.broadlink_name = null;
		this.broadlink_commandName = null;
		this.broadlink_repeatCount = 0;
		this.itach_name = null;
		this.itach_connectorAddress = null;
		this.itach_commandShortName = null;
		this.itach_repeatCount = 0;
		this.vera_name = null;
		this.vera_deviceNum = 0;
		this.vera_service = VeraService.DimmerValue;
		this.vera_value = null;
		this.hass_servername = null;
		this.hass_entityid = null;
		this.hass_method = null;
		this.hass_value = null;
	}
}